os/fs/smartfs/smartfs_utils.c

```diff
@@ -1387,6 +1380,7 @@ int smartfs_writeentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s new_
 
 	memset(direntry->name, 0, fs->fs_llformat.namesize + 1);
 	strncpy(direntry->name, filename, fs->fs_llformat.namesize);
+	direntry->flags = new_entry.flags;
```

Comment:
regarding direntry->name here, you gonna update in next PR, right?
at that time please consider we have to reveal direntry here.. I think we can do this in previous function as per I told you.
Then some of parameter also need to modified.

Reply:
Yes, changes for allocation or direntry->name in the next PR.